2765. Longest Alternating Subarray/solution.py

```diff
@@ -0,0 +1,49 @@
+class Solution:
+
+	# given a 0-indexed int array, find the maximum length alternating subarray
+	# where:
+	
+	# 1. alternating means the subarray goes [x, x+1, x, x+1...] for the 
+	# entire subarray
+	# 2. the array is between 2 and 100 in length
+	# 3. ints in the array are equal to or between 1 and 10^4
+    def alternatingSubarray(self, nums: List[int]) -> int:
+
+    	# declare a subArraySize variable that keeps track of length of 
+    	# alternating subarrays.
+
+    	# declare a list to hold potential subArraySizes.
+
+    	# loop through the array starting from 0, increment index by 1 
+    	# each time. this loop is the outer loop.
+
+    		# if the int at next index is one greater than int at the 
+    		# current index:
+
+    			# set subArraySize = 2 since we have an alternating subarray 
+    			# pair.
+
+    			# create a new variable subArrayIndex and set that equal to the 
+    			# current outer loop index + 1.
+
+    			# check for continuation of the subarray via:
+
+    			# while current value (at subArrayIndex) - 1 = next value and 
+    			# current value = next next value and subArrayIndex 
+    			# (the current index) is less than length of nums - 2:
+
+    				# increment subArraySize by 2
+    				# increment subArrayIndex by 2
+    			
+    			# reached end of sub array. append subArraySize to the list of 
+    			# subArraySizes.
+    			
+    			# add subArraySize - 1 to outer loop index so we skip the 
```

Comment:
While this works, consider that you have an outer loop with a hole in it (when you add subArraySize - 1 to the outer loop index) and an inner loop that fits almost entirely in that hole. Do you think you can use a variable to keep track of some state that would let you collapse the two loops into one? You don't actually need to change your solution as you exercise your coding skills just as much when you code according to your solution. Just something to think about.